src/targets/gpu/deconvolution.cpp

```diff
@@ -9,15 +9,32 @@ namespace gpu {
 shape miopen_deconvolution::compute_shape(const std::vector<shape>& inputs) const
 {
     check_shapes{inputs, *this}.has(4).standard();
-    return op.compute_shape({inputs.at(0), inputs.at(1)});
+    std::vector<shape> conv_inputs(inputs.begin(), inputs.begin() + 2);
+    check_shapes{conv_inputs, *this}.max_ndims(5);
+    return op.compute_shape(conv_inputs);
 }
+
+inline shape recompute_shape_to_2d(const shape& input)
```

Comment:
How about the name reshape_to_2d() here?

Reply:
I might rename it to reshape_if_1d to make it more clear that this is only done if the op is 1d